feat: complete issue #1 with maven

Juliapp123 opened the pull request:
closes #1 . we added code for printing the cloned repo aswell, just to make sure it works. Can be removed later if not needed.

Commit:
feat: complete issue #1

RasmussBrodin committed:
feat: complete #10

Changed to use maven, will be easier for testing i think.

VincentLagerros commented on src/main/java/se/kth/ContinuousIntegrationServer.java:
```diff
@@ -61,16 +65,52 @@ public void handle(String target,
+        File folder = new File(buildDirectory);
+        File[] listOfFiles = folder.listFiles();
+        if (listOfFiles != null) {
+            for (int i = 0; i < listOfFiles.length; i++) {
```
`for (File listOfFile : listOfFiles) {`

VincentLagerros commented on src/main/java/se/kth/ContinuousIntegrationServer.java:
```diff
@@ -61,16 +65,52 @@ public void handle(String target,
         }
     }
+
+    static void printRepo(String buildDirectory){
```
`throws SecurityException`

RasmussBrodin replied:
We used this to test and see if we got the correct repo. It is from StackOverflow, so it should not be included in our final submission. Let's just leave it for now, and remove it completely when we don't need it anymore?

VincentLagerros commented on src/main/java/se/kth/ContinuousIntegrationServer.java:
```diff
@@ -61,16 +65,52 @@ public void handle(String target,
+        Process process = new ProcessBuilder("mvn", "compile")
+            .directory(new File(buildDirectory)) // same as  "cd .serverbuild mvn compile" have to go to the correct dir
+            .start();
+        if (process.waitFor() != 0) {
```
I expect this pattern to be common when using ProcessBuilder, so maybe make it a function.

VincentLagerros commented on src/main/java/se/kth/ContinuousIntegrationServer.java:
```diff
@@ -61,16 +65,52 @@ public void handle(String target,
+    /**
+     * Compiles the cloned repository
+     *
+     * @param file      the cloned file to compile
```
`@param file` No argument?

VincentLagerros reviewed (changes requested):
I would include .DS_Store in the .gitignore to avoid pushing them upstream. Your code looks fine, but I also included some clean code practices.

RasmussBrodin committed:
Fix: Extended gitignore and created function for processBuilder. Issue #1

- Added DS_Store gitignore

- Created startProcess() to handle ProcessBuilder execution in a directory.

Closes #1

RasmussBrodin committed:
Fix: Cloning uses startProcess(). Issue #1

Pair Programming with Vincent.

Closes #1

VincentLagerros reviewed (approved):
Very good!
